### ISSUE TITLE
I found irreversible signal was trigger twice when replay !!!

### ISSUE THREAD

**Issue by firesWu** [EOSIO/eos#6794]
I found that two places triggered the irreversible signal.
```
forkdatabase.cpp  fork_database::prune( const block_state_ptr& h )
auto itr = my->index.find( h->id );
if( itr != my->index.end() ) {
 irreversible(*itr);
 ilog("${bn} ${block}",("bn",h->block_num)("block",h->dpos_irreversible_blocknum));
 my->index.erase(itr);
}

controller.cpp  replay_push_block( const signed_block_ptr& b, controller::block_status s ) 
// on replay irreversible is not emitted by fork database, so emit it explicitly here
if( s == controller::block_status::irreversible ){
	emit( self.irreversible_block, new_header_state );
        ilog("${bn} ${block}",("bn",new_header_state->block_num)("block",new_header_state->dpos_irreversible_blocknum));
}

history_plugin.cpp
my->irreversible_block_connection.emplace(
	chain.irreversible_block.connect( [&]( const chain::block_state_ptr& bs ) {
	  ilog("${result}",("result",bs->block_num));
	} ));

```
so I add ilog to print. 
In versione 1.6.0
I found that they all print log and history_plugin was print twice. so this comment will wrong, it will emit by fork database on replay.

In version 1.2.0
two place was all print log, but history_plugin was print once.  `irreversible(*itr);` was exec , but it doesn't work.

**Comment by heifner** [comment]
Thanks for the report. Verified signal called multiple times:

```
info  2019-02-21T13:50:55.009 thread-0  history_plugin.cpp:364        operator()           ] block 600, 000002582a96d0d231f383a7f6bb13586c989cb24e407d58de8a8ba777bb4b7f
info  2019-02-21T13:50:55.009 thread-0  history_plugin.cpp:364        operator()           ] block 602, 0000025af641e7df4c0c43d17d487e378f7f9235a52eee579c23214a70af284b
info  2019-02-21T13:50:55.009 thread-0  history_plugin.cpp:364        operator()           ] block 601, 00000259c8cb6b4307850f8f2a9553693d1775172e55296c5005e96dd5d8d0d3
info  2019-02-21T13:50:55.009 thread-0  history_plugin.cpp:364        operator()           ] block 603, 0000025b6dda0e6c3e7773a6314186b47f79d9ddc52e23c5dc9be418492439db
info  2019-02-21T13:50:55.009 thread-0  history_plugin.cpp:364        operator()           ] block 602, 0000025af641e7df4c0c43d17d487e378f7f9235a52eee579c23214a70af284b
info  2019-02-21T13:50:55.009 thread-0  history_plugin.cpp:364        operator()           ] block 604, 0000025c0d9f50b46a828cd18f1c20df843c933c986180f233b2de1f5a257805
info  2019-02-21T13:50:55.009 thread-0  history_plugin.cpp:364        operator()           ] block 603, 0000025b6dda0e6c3e7773a6314186b47f79d9ddc52e23c5dc9be418492439db
info  2019-02-21T13:50:55.009 thread-0  history_plugin.cpp:364        operator()           ] block 605, 0000025dcb4af3ba073ab957b5a9b95d6f3fd25ca114e9f98867e24af06b2f69
info  2019-02-21T13:50:55.009 thread-0  history_plugin.cpp:364        operator()           ] block 604, 0000025c0d9f50b46a828cd18f1c20df843c933c986180f233b2de1f5a257805
info  2019-02-21T13:50:55.010 thread-0  history_plugin.cpp:364        operator()           ] block 606, 0000025ef62135923ef59ba7c6833c09a051311e22f012f40d2000a368b047cd
info  2019-02-21T13:50:55.010 thread-0  history_plugin.cpp:364        operator()           ] block 605, 0000025dcb4af3ba073ab957b5a9b95d6f3fd25ca114e9f98867e24af06b2f69
```

**Comment by arhag** [comment]
PR #6624 has fixed this issue.

**Comment by crazybits** [comment]
fyi, as tested, v1.7 do not solve issue

**Comment by jgiszczak** [comment]
#6624 has not yet been merged to a release track branch.  It is not in v1.7.0-rc1.

**Comment by heifner** [comment]
Fixed in 1.8.
